langroid/language_models/openai_gpt.py

```diff
@@ -50,37 +50,52 @@ class OpenAICompletionModel(str, Enum):
     TEXT_DA_VINCI_003 = "text-davinci-003"  # deprecated
     TEXT_ADA_001 = "text-ada-001"  # deprecated
     GPT4 = "gpt-4"  # only works on chat-completion endpoint
-    LOCAL = "local"  # dummy for any local model
+
+
+_context_length: Dict[str, int] = {
+    # can add other non-openAI models here
+    OpenAIChatModel.GPT3_5_TURBO: 4096,
+    OpenAIChatModel.GPT4: 8192,
+    OpenAIChatModel.GPT4_NOFUNC: 8192,
+    OpenAICompletionModel.TEXT_DA_VINCI_003: 4096,
+}
+
+_cost_per_1k_tokens: Dict[str, Tuple[float, float]] = {
+    # can add other non-openAI models here.
+    # model => (prompt cost, generation cost) in USD
+    OpenAIChatModel.GPT3_5_TURBO: (0.0015, 0.002),
+    OpenAIChatModel.GPT4: (0.03, 0.06),  # 8K context
+    OpenAIChatModel.GPT4_NOFUNC: (0.03, 0.06),
+}
 
 
 class OpenAIGPTConfig(LLMConfig):
+    """
+    Class for any LLM with an OpenAI-like API: besides the OpenAI models this includes:
+    (a) locally-served models behind an OpenAI-compatible API
```

Comment:
Should we use it also for Azure OpenAI? though there are slight modifications like `engine` instead of `model`

Reply:
Also, do we need still to implement the context stuff? I think this class can serve the same goal.

Reply:
for azure we could leave our code as is, and also of course it will be feasible via litellm. I would guess litellm handles the `Engine/model` variant in their proxy

Reply:
> the context stuff?

what are you referring to?

Reply:
this issue https://github.com/langroid/langroid/issues/233 and this PR https://github.com/langroid/langroid/pull/244